### ISSUE TITLE
[PRO] RedisStorage support for setting TTL on redis subscription key.

### ISSUE THREAD

**Issue by krapans** [rmosolgo/graphql-ruby#3252]
**Is your feature request related to a problem? Please describe.**

Currently none of subscription keys contain TTL value on redis key. Which means it can lead into a situation where there are stale keys in your redis which eats redis memory.

**Describe the solution you'd like**

Currently when setting a subscription key, it doesn't accept a scenario where we could set TTL for those keys, even if it's some ridiculous default one like _2 months_.

```ruby
# redis_storage.rb
def write_subscription(*)
  # ...
  # line 39
  @redis.set(key, value)
  # ...
end
```

By default redis set accepts extra optional parameters where one of them is TTL.

https://github.com/redis/redis-rb/blob/master/lib/redis.rb#L830

**Describe alternatives you've considered**

Alternative would be to refactor `write_subscription` into a bit more extensible logic where anyone can patch/modify original flow with extra functionality that they see as a solution for their problem.

**Additional context**

Currently the workaround which we use is extending logic of redis storage by inheritance and adding extra logic which is setting TTL for subscription redis key manually. This results in extra call towards redis which could be mitigated if we introduce an option adding this upon setting subscription key in redis.

**Comment by rmosolgo** [comment]
Hey, thanks for the suggestion. I'll take a crack at adding it and follow up here. 

In the meantime, I'm curious, have you seen the buildup of stale data in your system? It's _meant_ to clean up old data in two ways: 

- By receiving webhooks from the transport provider (Pusher or Ably) 
- Whenever running a subscription update, it checks with the transport provider if the client is still connected; if it isn't, then it deletes the subscription. 

However, it's possible that neither of those would clean up subscriptions when the webhook is _missed_ and then no subscription is triggered that would be delivered to the terminated subscription. I wonder if that's happening 🤔

**Comment by krapans** [comment]
Hey, currently in our case this happens as we are using a custom websocket server for a specific problem solution. And in some scenarios there's some problem with handling subscription key destruction in redis and most probably client newer tries to connect to the same subscription scope again which leads to old subscriptions in redis.
We still need to look into why this happens overall that hook from our custom websocket server is not received/missed sometimes.

To mitigate this situation for now, TTL on subscription keys will deal with these edge cases and clean up those old subscriptions eventually.

**Comment by rmosolgo** [comment]
Thanks again for this suggestion. I just released graphql-pro 1.16.2 which includes a `stale_ttl_s:` option for leveraging Redis's TTL like you mentioned above. You can find it in the docs here: https://graphql-ruby.org/subscriptions/pusher_implementation.html#schema-configuration (And here, same docs: https://graphql-ruby.org/subscriptions/ably_implementation.html#schema-configuration) 

Give it a shot and let me know how it goes!
